internal/uvm/memory.go

```diff
@@ -0,0 +1,45 @@
+package uvm
+
+import (
+	"context"
+	"fmt"
+
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const (
+	bytesPerPage = 4096
+	bytesPerMB   = 1024 * 1024
+)
+
+// UpdateMemory makes a call to the VM's orchestrator to update the VM's size in MB
+func (uvm *UtilityVM) UpdateMemory(ctx context.Context, sizeInMB uint64) error {
+	req := &hcsschema.ModifySettingRequest{
+		ResourcePath: memoryResourcePath,
+		Settings:     sizeInMB,
+	}
+	if err := uvm.modify(ctx, req); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetAssignedMemoryInMB returns the amount of assigned memory for the UVM in MBs
+func (uvm *UtilityVM) GetAssignedMemoryInMB(ctx context.Context) (uint64, error) {
```

Comment:
Depending on use case, it might be better to just remember how much memory is assigned to a UVM, rather than needing to query it from the HCS each time.

Reply:
I agree, though this is placed here mostly for testing purposes since hcsSystem is not exported.